src/librustdoc/html/static/main.js

```diff
@@ -502,54 +597,92 @@
         }
 
         function initSearchNav() {
-            var hoverTimeout, $results = $('.search-results .result');
+            var hoverTimeout;
 
-            $results.on('click', function() {
-                var dst = $(this).find('a')[0];
+            var click_func = function(e) {
+                var el = e.target;
+                // to retrieve the real "owner" of the event.
+                while (el.tagName !== 'TR') {
```

Comment:
Jquery allows to get the real owner of the event, whereas you have to find yourself normally. So if you bind something on an element, if it has children, then it's certainly one of the children which will "emit" the event.